File: VerUpdater/trunk/source/console/console/src/main/java/com/ebeijia/module/system/service/systemToken/SysTokenServiceImpl.scala
package com.ebeijia.module.system.service.systemToken

import com.ebeijia.util.common.EncryptMD5Util
import org.ebeijia.tools.UUID4J
import org.springframework.cache.annotation.Cacheable
import org.springframework.stereotype.Service

/**
 * Token
 * @author zhicheng.xu
 *         16/5/24
 */
@Service
final class SysTokenServiceImpl extends SysTokenSerivce{

  /**
   * 设置token
   * @param id
   * @return
   */
  @Cacheable(value = Array("sysToken"), key = "#root.method.name+#id")
  def setToken( id :String) :String={
    val uuid: String = UUID4J.getUUID + id
    val token: String = EncryptMD5Util.encrypt(uuid)
    token
  }

//  def getToken(id: String, token: String): Boolean = {
//    val newToken: String = this.setToken(id)
//    var flag: Boolean = false
//    if (newToken.equals(token)) {
//      flag = true
//    }
//    flag
//  }
}

File: VerUpdater/trunk/source/web/VerUpdaterSvr/core/src/main/java/com/ebeijia/service/system/SysTokenServiceImpl.java
package com.ebeijia.service.system;

import org.springframework.stereotype.Service;

import com.ebeijia.common.PackInspector;

@Service
public class SysTokenServiceImpl implements SysTokenSerivce {
	public static final String conToken = "QYVD8W2JJW4KYDP2YMKWFX36";

	@Override
	public Boolean judgeToken(String token) {
		Boolean flag = false;
		if (token != null && token.equals(conToken)) {
			flag = true;
		}
		return flag;
	}

	@Override
	public Boolean judgeSign(String sendTime, String reqData,String sign) {
		String newSign = "sendTime=" + sendTime + "data=" + reqData;
		System.out.println(newSign);

		return PackInspector.getInstances().md5CompareByBuffer(newSign,sign,false);
	}

}
